tests/v2/validation/rbac/secrets/rbac_secrets_test.go

```diff
@@ -0,0 +1,362 @@
+			standardUserContext, err := standardUserClient.WranglerContext.DownStreamClusterWranglerContext(rbs.cluster.ID)
+			require.NoError(rbs.T(), err)
+			secretList, err := standardUserContext.Core.Secret().List(namespace.Name, metav1.ListOptions{})
+			switch tt.role.String() {
+			case rbacaction.ClusterOwner.String(), rbacaction.ProjectOwner.String(), rbacaction.ProjectMember.String():
+				require.NoError(rbs.T(), err, "failed to list secret")
+				require.Equal(rbs.T(), len(secretList.Items), 1)
+				require.Equal(rbs.T(), secretList.Items[0].Name, createdSecret.Name)
+			case rbacaction.ClusterMember.String(), rbacaction.ReadOnly.String():
+				require.Error(rbs.T(), err)
+				require.True(rbs.T(), errors.IsForbidden(err))
+			}
+		})
+	}
+}
+
+func (rbs *RbacSecretTestSuite) TestUpdateSecret() {
+	subSession := rbs.session.NewSession()
+	defer subSession.Cleanup()
+
+	tests := []struct {
+		role   rbacaction.Role
+		member string
+	}{
+		{rbacaction.ClusterOwner, rbacaction.StandardUser.String()},
+		{rbacaction.ClusterMember, rbacaction.StandardUser.String()},
+		{rbacaction.ProjectOwner, rbacaction.StandardUser.String()},
+		{rbacaction.ProjectMember, rbacaction.StandardUser.String()},
+		{rbacaction.ReadOnly, rbacaction.StandardUser.String()},
+	}
+
+	for _, tt := range tests {
+		rbs.Run("Validate updating secret as user with role "+tt.role.String(), func() {
+			log.Info("Create a project and a namespace in the project.")
+			adminProject, namespace, err := projectaction.CreateProjectAndNamespace(rbs.client, rbs.cluster.ID)
+			require.NoError(rbs.T(), err)
+
+			log.Infof("Create a standard user and add the user with role %s", tt.role)
+			newUser, standardUserClient, err := rbacaction.CreateUserAndAddToRole(rbs.client, tt.member, tt.role.String(), rbs.cluster, adminProject)
+			require.NoError(rbs.T(), err)
+			rbs.T().Logf("Created user: %v", newUser.Username)
+
+			log.Infof("As a %v, create a secret in the project %v", rbacaction.Admin, adminProject.Name)
+			createdSecret, err := secretaction.CreateSecret(rbs.client, rbs.cluster.ID, namespace.Name)
+			require.NoError(rbs.T(), err, "failed to create secret")
+
+			log.Infof("As a %v, update the secret.", tt.role.String())
+			standardUserContext, err := standardUserClient.WranglerContext.DownStreamClusterWranglerContext(rbs.cluster.ID)
+			require.NoError(rbs.T(), err)
+
+			newData := map[string][]byte{
+				"foo": []byte("bar"),
+			}
+			updatedSecretObj := secretaction.UpdateSecretData(createdSecret, newData)
+			updatedSecret, err := standardUserContext.Core.Secret().Update(updatedSecretObj)
+			switch tt.role.String() {
+			case rbacaction.ClusterOwner.String(), rbacaction.ProjectOwner.String(), rbacaction.ProjectMember.String():
+				require.NoError(rbs.T(), err, "failed to update secret")
+				require.NotNil(rbs.T(), updatedSecret)
+				require.Contains(rbs.T(), updatedSecret.Data, "foo")
+				require.Equal(rbs.T(), updatedSecret.Data["foo"], []byte("bar"))
```

Comment:
assert might be better here. if its require.Equal and when the conditions fails, the tests wont run other validations after this.

Reply:
Great point, modified